project/cmake/modules/FindCXX11.cmake

```diff
@@ -0,0 +1,18 @@
+include(TestCXXAcceptsFlag)
+
+# try to use compiler flag -std=c++11
+check_cxx_accepts_flag("-std=c++11" CXX_FLAG_CXX11)
+if(CXX_FLAG_CXX11)
+  add_options (CXX ALL_BUILDS "-std=c++11")
+  set(CXX_STD11_FLAGS "-std=c++11")
+else()
```

Comment:
Thanks for pointing that out. There are a few ways to do that in CMake. Do we have minimum compiler versions defined somewhere? Because even if a compiler has the `-std=c++11` flag, it doesn't mean that it support all C++11 features that are used in the project. And Microsoft compilers don't have the flag at all. There's also a way to specify needed C++ features and CMake figures out the options (https://cmake.org/cmake/help/latest/manual/cmake-compile-features.7.html). But that might be a bigger task considering the project size.